### ISSUE TITLE
Issue when destroying the grid with custom React components

### ISSUE THREAD

**Issue by chuckn0rris** [bryntum/support#1339]
[Forum post](https://www.bryntum.com/forum/viewtopic.php?f=51&t=14909&p=75601)
Error appears on initiating grid destroy - as example after switch tabs from grid to something else. 
But, react cell editor must be used before.

Reproducible with react/javascript/basic example for grid. 

Steps to reproduce: 
1. add router to make possible initiate grid destroy.
2. double-click on "Original 6 (react editor)" cell to show the editor.
3. click to initiate router page change.

An error appeared:

<img width="1061" alt="Screenshot 2020-08-14 at 11 47 51" src="https://user-images.githubusercontent.com/7203098/90231526-23031f80-de3d-11ea-81ac-b4a13f5ab385.png">

This code might be used for `render()` function in `Main.js` to reproduce it fast:

```
    /**
     * render method
     */
    render = () => {
        return (
        <Router>
            <div>
                <ul>
                    <li>
                        <Link to="/">Grid</Link>
                    </li>
                    <li>
                        <Link to="/home">Home</Link>
                    </li>
                </ul>
                <hr />

                <Switch>
                    <Route exact path="/">
                         // all inside <Fragment> tag - is just copy-paste from original example renderer.                        
                         <Fragment>
                            <Header
                                selectedRow={this.state.selectedRow}
                                headerUrl='..'
                                rowHeight={this.state.rowHeight}
                                onRowHeightChange={this.handleRowHeightChange}
                                onAddClick={this.handleAddClick.bind(this)}
                                onRemoveClick={this.handleRemoveClick.bind(this)}
                            />
                            <BryntumGrid
                                ref={'grid'}

                                // Make grid grow to fit rows, not recommended since it has a negative impact on performance
                                // autoHeight={true}

                                // Initial row height
                                rowHeight={this.state.rowHeight}

                                // Initial grouping, with custom group renderer
                                groupFeature={{
                                    field : 'division',
                                    groupRenderer(data) {
                                        if (data.column === data.grid.columns.first) {
                                            return data.groupColumn.text + ': ' + data.groupRowFor;
                                        }

                                        return '';
                                    }
                                }}

                                // Columns in grid
                                columns={[
                                    {
                                        field : 'team',
                                        text : 'Team<div class="small-text">(inline JSX)</div>',
                                        flex : 1,
                                        // Using simple inline JSX
                                        renderer: ({ value }) => <b>{value}</b>
                                    },
                                    { field : 'city', text : 'City', flex : 1 },
                                    {
                                        field    : 'original',
                                        text     : 'Original 6<div class="small-text">(React editor)</div>',
                                        align    : 'center',
                                        width    : 120,
                                        renderer : ({ value }) => value ? 'Yes' : 'No',
                                        editor   : ref => <DemoEditor ref={ref}/>
                                    },
                                    { field : 'division', text : 'Division', flex : 1, hidden : true },
                                    { field : 'conference', text : 'Conference', flex : 1 },
                                    {
                                        field    : 'team',
                                        text     : 'Cheer<div class="small-text">(React component)</div>',
                                        width    : 160,
                                        editor   : false,
                                        // Using custom React component
                                        renderer : ({ record }) =>
                                            <DemoButton text={'Go ' + record.team + '!'}
                                                        onClick={() => this.handleCellButtonClick(record)}/>
                                    }
                                ]}

                                // Rows to display in grid
                                data={this.state.data}

                                // Sync data instead of replacing entire dataset
                                syncDataOnLoad={true}

                                onSelectionChange={this.handleSelectionChange}
                            />

                        </Fragment>
                    </Route>
                    <Route path="/home">
                        <Fragment />
                    </Route>
                </Switch>
            </div>
        </Router>
        );
    }; // eo function render
```
Do not forget to add import: 
```
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link
  } from "react-router-dom";
```

**Comment by isglass** [comment]
Looks like we should `try catch` that delete, we cannot be sure we can remove whats there

**Comment by jsakalos** [comment]
This issue has been resolved in `4.1.0`
